### ISSUE TITLE
Data views filter chips color contrast below standards

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#62760]
### Description

The  buttons to filter patterns according to synced status are below color contrast. The button is at 4.04:1, where the minimum contrast is 4.5:1.

See previous relevant issue: https://github.com/WordPress/gutenberg/issues/52555. That issue was with a different wrapping design, but the same control group.

I'll also observe that these should be real buttons, rather than `div` elements with events and tabindex; but that's a separate issue.

### Step-by-step reproduction instructions

1. Go to Patterns
2. Choose All Patterns
3. Inspect the 'Sync Status' button colors

### Screenshots, screen recording, code snippet

![image](https://github.com/WordPress/gutenberg/assets/1686451/0125fddb-0948-427a-856d-27b9d19e8223)
![image](https://github.com/WordPress/gutenberg/assets/1686451/fe02f870-64d6-44ce-a94f-15e207c16f9c)

### Environment info

- WordPress 6.5.4
- Gutenberg build from main branch 6/21/2024

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
@jasmussen @jameskoster as a heads up. If we can get this fixed for 6.6, let's add it to the board.

**Comment by jasmussen** [comment]
Seems like we can either use $gray-800 for the text, resulting in this:

![Screenshot 2024-06-26 at 08 42 23](https://github.com/WordPress/gutenberg/assets/1204802/fd7e524e-be91-4a33-9401-a83966f08e3f)

That would match these chips:
![Screenshot 2024-06-26 at 08 42 37](https://github.com/WordPress/gutenberg/assets/1204802/efc2931a-edf8-4244-a08b-a8a9c51296d9)

Or we can use a border instead of a background:
![Screenshot 2024-06-26 at 08 43 34](https://github.com/WordPress/gutenberg/assets/1204802/2077ec33-0096-4749-b6f3-b4c12ca7663e)

I'd love a quick check by @jameskoster before I make either PR, as the interface is meant for additional places.
